joins.sql

```diff
@@ -0,0 +1,105 @@
+SELECT posts.title AS post_title, posts.url AS post_url, comments.body AS comment_body
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id
+WHERE posts.created_at < '2015-01-01 00:00:00';
+
+-- 9. Create a query to get the all rows in the comments table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the post was created after January 1, 2015
+SELECT posts.title AS post_title, posts.url AS post_url, comments.body AS comment_body
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id
+WHERE posts.created_at > '2015-01-01 00:00:00';
+
+--10. Create a query to get the all rows in the comments table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the comment body contains the word 'USB'
+SELECT posts.title AS post_title, posts.url AS post_url, comments.body AS comment_body
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id
+WHERE comments.body LIKE '%USB%';
+
+--11. Create a query to get the post title (aliased as post_title), first name of the author of the post, last name of the author of the post, and comment body (aliased to comment_body), where the comment body contains the word 'matrix' ( should have 855 results )
+SELECT posts.title AS post_title, users.first_name, users.last_name, comments.body AS comment_body
+FROM comments
+    INNER JOIN posts
+    ON comments.posts_id = posts.id
+    INNER JOIN users
+    ON posts.users_id = users.id
+WHERE comments.body LIKE '%matrix%';
+
+--12. Create a query to get the first name of the author of the comment, last name of the author of the comment, and comment body (aliased to comment_body), where the comment body contains the word 'SSL' and the post content contains the word 'dolorum' ( should have 102 results )
+SELECT users.first_name, users.last_name, comments.body AS comment_body
+FROM comments
+    INNER JOIN posts
+    ON comments.posts_id = posts.id
+    INNER JOIN users
+    ON posts.users_id = users.id
+WHERE comments.body LIKE '%SSL%' AND posts.content LIKE '%dolorum%';
+
+--13. Create a query to get the first name of the author of the post (aliased to post_author_first_name), last name of the author of the post (aliased to post_author_last_name), the post title (aliased to post_title), username of the author of the comment (aliased to comment_author_username), and comment body (aliased to comment_body), where the comment body contains the word 'SSL' or 'firewall' and the post content contains the word 'nemo' ( should have 218 results )
+SELECT users.first_name AS post_author_first_name, users.last_name AS post_author_last_name, posts.title AS post_title, users.username AS comment_author_username, comments.body AS comment_body
```

Comment:
Although the post author first and last name are right here, the commentor's username is wrong. You'll need to join on the users table twice to get that information.